AuthenticationAndAuthorization/DataAccounts/Repositories/UserRepository.cs

```diff
@@ -0,0 +1,72 @@
+﻿using DataAccounts.Repositories;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccounts
+{
+    public class UserRepository : IUserRepository
+    {
+        private readonly UserManager<User> _userManager;
+        private readonly SignInManager<User> _signInManager;
+        private readonly IUnitOfWork _unitOfWork;
+
+        public UserRepository(UserManager<User> userManager, SignInManager<User> signInManager, IUnitOfWork unitOfWork)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task Register(User user, string password, string role)
+        {
+            await _unitOfWork.BeginTransactionAsync();
+
+            try
+            {
+                var result = await _userManager.CreateAsync(user, password);
+
+                if (result.Succeeded)
+                {
+                    var roleResult = await _userManager.AddToRoleAsync(user, role);
+
+                    if (roleResult.Succeeded)
+                    {
+                        await _unitOfWork.CommitAsync();
+                    }
+                    else
+                    {
+                        throw new Exception("Failed to assign role to user.");
+                    }
+                }
+                else
+                {
+                    throw new Exception("Failed to create user.");
+                }
```

Comment:
Throw concrete exceptions, throwing general `Exception` is bad practice. You may create your own exception or use system ones